pallets/staking_rewards/src/lib.rs

```diff
@@ -0,0 +1,235 @@
+}
+
+impl<T: Config> Module<T> {
+    // TODO: Needed as RPC?
+    /// Compute emission reward of an era. It considers the remaining emission supply and the decay in
+    /// addition to NPoS inflation. Returns the emission reward and the reduced emission supply after
+    /// emitting the rewards.
+    fn emission_reward_for_era(
+        reward_curve: &PiecewiseLinear,
+        total_staked: BalanceOf<T>,
+        total_issuance: BalanceOf<T>,
+        era_duration_millis: u64,
+    ) -> (BalanceOf<T>, BalanceOf<T>) {
+        let emission_supply = Self::staking_emission_supply();
+        if !Self::staking_emission_status() {
+            return (BalanceOf::<T>::zero(), emission_supply);
+        }
+
+        // Emission reward for the whole year
+        let yearly_rewards = Self::get_yearly_emission_reward(
+            reward_curve,
+            total_staked,
+            total_issuance,
+            emission_supply,
+        );
+
+        // Emission reward for the era
+        let emission_reward =
+            Self::get_emission_reward_for_era_given_yearly(era_duration_millis, yearly_rewards);
+
+        let remaining = emission_supply.saturating_sub(emission_reward);
+        (emission_reward, remaining)
+    }
+
+    // TODO: Needed as RPC?
+    /// Get yearly emission reward as per NPoS and remaining emission supply. The reward is taken
+    /// from remaining emission supply and is proportional to the ratio of current NPoS inflation to
+    /// maximum NPoS inflation
+    fn get_yearly_emission_reward(
+        reward_curve: &PiecewiseLinear,
+        total_staked: BalanceOf<T>,
+        total_issuance: BalanceOf<T>,
+        emission_supply: BalanceOf<T>,
+    ) -> BalanceOf<T> {
+        let reward_proportion_of_max = Self::get_yearly_emission_reward_prop_as_per_npos_only(
+            reward_curve,
+            total_staked,
+            total_issuance,
+        );
+        let yearly_emission = Self::get_max_yearly_emission(emission_supply);
+        reward_proportion_of_max * yearly_emission
+    }
+
+    /// Get proportion of NPoS emission as per current staking rate (`total_staked` / `total_issuance`) to
+    /// emission as per ideal staking rate. Doesn't consider remaining emission supply
+    fn get_yearly_emission_reward_prop_as_per_npos_only(
+        reward_curve: &PiecewiseLinear,
+        total_staked: BalanceOf<T>,
+        total_issuance: BalanceOf<T>,
+    ) -> Perbill {
+        let reward_as_per_npos = Self::get_yearly_emission_reward_as_per_npos_only(
+            reward_curve,
+            total_staked,
+            total_issuance,
+        );
+        let reward_proportion_of_max =
+            Perbill::from_rational(reward_as_per_npos, reward_curve.maximum * total_issuance);
+        reward_proportion_of_max
+    }
+
+    /// Get emission per year according to NPoS as described in token economics doc here
+    /// https://research.web3.foundation/en/latest/polkadot/overview/2-token-economics.html. Doesn't
```

Comment:
this is super helpful 👍 for visualizing the staking rewards curve (with the graph they have on that page)

Could we add a README to the staking rewards pallet ?
With 
- this link
- a short summary of what our staking tokenomics are
- and maybe a copy of the graph (there's one image on the page that has the top rewards point at 75%)